p2p/server/msgserver.go

```diff
@@ -162,34 +164,40 @@ func (p *MessageServer) Close() {
 
 // readLoop reads incoming messages and matches them to requests or responses.
 func (p *MessageServer) readLoop(ctx context.Context) error {
+	logger := p.WithContext(ctx)
 	sctx := log.WithNewSessionID(ctx)
 	timer := time.NewTicker(p.requestLifetime + time.Millisecond*100)
 	defer timer.Stop()
-	defer p.With().Info("shutting down protocol", log.String("protocol", p.name))
+	defer logger.With().Info("shutting down protocol", log.String("protocol", p.name))
 	for {
 		select {
 		case <-ctx.Done():
 			return ctx.Err()
 		case <-timer.C:
-			p.tg.Go(func(ctx context.Context) error {
+			if err := p.tg.Go(func(ctx context.Context) error {
 				p.cleanStaleMessages()
 				return nil
-			})
+			}); err != nil {
+				logger.With().Panic("error from cleanStaleMessages taskgroup", log.Err(err))
```

Comment:
you shouldn't panic here. you can panic only in initializer, because stop can't be executed concurrently with initializer. in this case this is valid for tg.Go to return an error that tg was terminated.